### PR TITLE
Add DSP energy pre-filter for Silero VAD streaming

### LINKED ISSUE #101: Improve Silero VAD throughput with DSP pre-filter and benchmark
## Context

Analysis of [AtharvBhat/fast-vad](https://github.com/AtharvBhat/fast-vad) — a Rust DSP-only VAD using 8-band log-energy + logistic regression (no neural network). Key findings:

| System | Speed | AVA-Speech F1 | LibriVAD -5dB F1 |
|---|---|---|---|
| fast-vad (batch) | **721x** RT | 0.680 | **0.928** |
| Silero (ONNX) | ~1x RT | **0.738** | 0.878 |
| WebRTC VAD | 63x RT | 0.669 | 0.927 |

fast-vad trades ~5-8% F1 for 700x+ speed — but notably **beats Silero at low SNR** (-5dB) due to adaptive noise floor tracking.

## Goal

Improve our Silero VAD throughput on long-form audio without sacrificing accuracy, inspired by fast-vad's techniques.

### 1. Two-Stage VAD Pipeline (high value)

Add a fast DSP energy classifier as Stage 1 pre-filter before Silero:

```
Audio → [Stage 1: DSP energy bands + threshold] → candidate regions → [Stage 2: Silero] → final segments
```

- 8 log-spaced frequency bands via `vDSP_fft_zrip` (Accelerate)
- Adaptive noise floor estimation (EMA α=0.01 during silence)
- Noise-normalized energy features + simple threshold
- Skip Silero inference on frames clearly below noise floor
- Expected: skip 40-60% of frames on typical long-form audio (meetings, podcasts)

### 2. Noise Floor Tracking (high value)

Adaptive noise floor estimation that updates during confirmed silence frames. This is why fast-vad outperforms Silero at -5dB SNR — noise-normalized features adapt to changing background conditions. Could be used both as a pre-filter feature and to improve Silero's effective accuracy in noisy environments.

### 3. Logit-Space Threshold Comparison (low effort)

Convert onset/offset probability thresholds to logits once at init (`log(p/(1-p))`), compare against raw logits instead of computing sigmoid every chunk. Minor but free.

### 4. Streaming Pre-Filter Mode

Extend `StreamingVADProcessor` to support the two-stage approach:
- DSP pre-filter runs on every chunk (~0 cost)
- Silero only invoked when energy exceeds noise floor
- Maintain Silero LSTM state correctly across skipped frames (reset after long silence gaps)

### Datasets
- **AVA-Speech**: Real-world audio with music, noise, overlapping speech
- **LibriVAD**: LibriSpeech + noise at -5, 0, 5, 10, 15, 20 dB SNR levels
- Or equivalent test sets we can source/construct

### Metrics
- **F1 / Precision / Recall** at each SNR level
- **Throughput**: realtime multiplier (samples processed / wall time), batch and streaming
- **Latency**: per-chunk p50/p99 for streaming mode
- **Frames skipped**: % of frames where Silero inference was avoided by pre-filter

### Baselines to Compare
1. Current Silero VAD (MLX) — no pre-filter
2. Current Silero VAD (CoreML) — no pre-filter
3. Silero + DSP pre-filter (MLX)
4. Silero + DSP pre-filter (CoreML)
5. DSP-only (no Silero) — to measure pre-filter accuracy floor

### Hardware
- Apple M-series (M1/M2/M3/M4), report chip variant with results

## Not In Scope

- Replacing Silero with a pure DSP classifier (accuracy tradeoff not worth it for our use case)
- Rayon-style parallelism (we use GCD/Swift concurrency)
- Rewriting the FFT STFT (Silero's learned Conv1d STFT is more powerful)

## References

- https://github.com/AtharvBhat/fast-vad
- fast-vad benchmark methodology: AVA-Speech + LibriVAD datasets
- Silero VAD v5: https://github.com/snakers4/silero-vad

### PR DESCRIPTION
**Pull request by ivan-digital**
## Summary
- 8-band FFT energy pre-filter (vDSP/Accelerate) skips clearly-silent chunks without neural inference
- Adaptive noise floor via EMA, 10-chunk warmup, 2-chunk look-back replay on skip→invoke transitions
- LSTM state reset after 50 consecutive skips (~1.6s) to prevent stale state
- Enabled by default in `vad-stream`; disable with `--no-pre-filter`
- E2E verified: segments match within 1 chunk (32ms), **81% skip rate** on test audio

Closes #101

## Test plan
- [x] 9 new `EnergyPreFilterTests` (warmup, silence skip, tone detection, band coverage, noise floor convergence, reset, edge cases)
- [x] 5 new tests in `SileroVADTests` (`updateContextOnly`, `resetLSTMState`, pre-filter integration with/without, reset)
- [x] All 33 existing tests pass
- [x] E2E: `audio vad-stream test_audio.wav` — [5.18s–8.38s] with pre-filter vs [5.22s–8.38s] without
- [ ] Verify on longer meeting/podcast audio for skip rate and boundary accuracy

### CODE REVIEW

**Comment by ivan-digital** [comment]
Closing — pre-filter provides negligible benefit on real meeting audio (1-9% skip rate vs projected 40-60%). Dense speech content leaves almost no chunks to skip, and Silero is already 23x+ RT without it.
